pkg/util/delayed_scheduler/delayed_scheduler.go

```diff
@@ -0,0 +1,140 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package delayedscheduler implements a scheduler that can be used to schedule the execution of a function at a given time.
+// Similar to time.AfterFunc but it doesn't use more than one timer at a time.
+package delayedscheduler
```

Comment:
I fill that you can reuse the work queue package to implement the retry mechanism with the back off policy

Reply:
I wasn't aware of the existence of this package. The latest commit uses it